### ISSUE TITLE
Docker container is stuck on PG_dump with docker-ce version 18.03

### ISSUE THREAD

**Issue by BlackTurtle123** [docker-library/postgres#502]
The official docker container is stuck on PG_dump with docker-ce version 18.03.
When we upgrade docker-ce to 18.06 on the same environment the PG_dump doesnt become stuck anymore.
If we downgrade, the pg_dump is stuck again, on every environment.
If we run pg_dump inside a script, the script becomes broken, because the dump never finishes. 
If we run pg_dump in command line to std.out, an enter when it is stuck, let it continious.

**Comment by wglambert** [comment]
Can you give all the commands you ran and any relevant contextual information or files for reproducing the issue

**Comment by BlackTurtle123** [comment]
the command we ran was:
-pg_dumpall -c -Upostgres
I am not sure if I can provide the data inside the database, since it contains sensitive company data.
A pg_dumpall on an empty database would never be stuck.

We literally just entered the docker container with exec, and ran the pg_dumpall -c -Upostgres command, and it became often stuck in the middle of printing the text and we had to hit enter and it would continue.
We tried to keep the reproduce scenario so small as possible. So the basic command became stuck.
Will tomorrow in the company see to show an example output, ofcource with hidden data

**Comment by tianon** [comment]
Honestly, I think the best we can do is recommend you attempt to create a reliable reproducer and report this upstream -- there's not likely anything we've done in the image to cause this nor much we can change about the image to fix it.

Is it something you can reliably reproduce if you copy the data to another machine? (this would help rule out hardware or system issues)

Can you reproduce it with newer versions of PostgreSQL? (to help upstream determine whether this is something they've already fixed in a newer release)

Can you reproduce it outside Docker with a similar environment to help narrow down whether it's really Docker at fault here?

I notice that `pg_dumpall` has a `--verbose` flag, but I'm not sure whether it'll really give you much more useful information -- worth a try.

The TL;DR here is that you might have a solid issue on your hands, but I think the absolute most we could do in the context of the _Docker image_ is document it, and it seems to be a sufficiently niche issue that I'm not convinced that's entirely warranted?

**Comment by BlackTurtle123** [comment]
> Is it something you can reliably reproduce if you copy the data to another machine? (this would help rule out hardware or system issues)

Yes this is the case, tried on different vm's aswell real servers. 

> Can you reproduce it with newer versions of PostgreSQL? (to help upstream determine whether this is something they've already fixed in a newer release)

Yes, postgreSQL 10 aswell 11, alpine as non alpine have this behavior

> I notice that pg_dumpall has a --verbose flag, but I'm not sure whether it'll really give you much more useful information -- worth a try.

Nowhere it outputs more info, or any error or root cause, neither in the logging.

**Comment by tianon** [comment]
> Honestly, I think the best we can do is recommend you attempt to create a reliable reproducer and report this upstream -- there's not likely anything we've done in the image to cause this nor much we can change about the image to fix it.

> ...

> The TL;DR here is that you might have a solid issue on your hands, but I think the absolute most we could do in the context of the Docker image is document it, and it seems to be a sufficiently niche issue that I'm not convinced that's entirely warranted?

Unfortunately, I don't think there's much more we can do for you here (in https://github.com/docker-library/postgres) -- it _sounds_ like your issue is still environmental, but I really don't know, and the best we can do is recommend you try one of the PostgreSQL mailing lists, [the Docker Community Forums](https://forums.docker.com/), [the Docker Community Slack](https://blog.docker.com/2016/11/introducing-docker-community-directory-docker-community-slack/), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker) for further help debugging.  No matter where you end up going for further help, they're going to need a minimal reproducer, so I'd suggest working on something like that (especially something you could reproduce in a cloud provider like AWS or Digital Ocean to further distance the reproducer from your current environment to help remove variables).
